staging/src/k8s.io/kube-aggregator/pkg/apiserver/apiserver.go

```diff
@@ -307,6 +313,7 @@ func (s *APIAggregator) RemoveAPIService(apiServiceName string) {
 	}
 	s.GenericAPIServer.Handler.NonGoRestfulMux.Unregister(proxyPath)
 	s.GenericAPIServer.Handler.NonGoRestfulMux.Unregister(proxyPath + "/")
+	s.openAPIAggregator.removeApiServiceSpec(apiServiceName)
```

Comment:
Fixed. All public methods of openapiAggregator are now thread safe and I changed this to call the public method.